misc-utils/uuidd.c

```diff
@@ -638,52 +632,78 @@ int main(int argc, char **argv)
-		errx(EXIT_FAILURE, _("socket name too long: %s"), socket_path);
+}
 
-	if (!no_pid && !pidfile_path)
-		pidfile_path = UUIDD_PIDFILE_PATH;
+int main(int argc, char **argv)
+{
+	const char	*err_context = NULL;
+	char		*cp;
+	int		ret;
+
+	struct uuidd_cxt_t uuidd_cxt = { .timeout = 0 };
+	struct uuidd_options_t uuidd_opts = { .socket_path = UUIDD_SOCKET_PATH };
+
+	setlocale(LC_ALL, "");
+	bindtextdomain(PACKAGE, LOCALEDIR);
+	textdomain(PACKAGE);
+	close_stdout_atexit();
+
+	parse_options(argc, argv, &uuidd_cxt, &uuidd_opts);
+
+	if (strlen(uuidd_opts.socket_path) >= sizeof(((struct sockaddr_un *)0)->sun_path))
+		errx(EXIT_FAILURE, _("socket name too long: %s"), uuidd_opts.socket_path);
+
+	if (!uuidd_opts.no_pid && !uuidd_opts.pidfile_path)
+		uuidd_opts.pidfile_path = UUIDD_PIDFILE_PATH;
 
 	/* custom socket path and socket-activation make no sense */
-	if (s_flag && uuidd_cxt.no_sock && !uuidd_cxt.quiet)
+	if (uuidd_opts.s_flag && uuidd_cxt.no_sock && !uuidd_cxt.quiet)
 		warnx(_("Both --socket-activation and --socket specified. "
 			"Ignoring --socket."));
 
-	if (num && do_type) {
-		ret = call_daemon(socket_path, do_type + 2, buf,
-				  sizeof(buf), &num, &err_context);
+	if (uuidd_opts.num && uuidd_opts.do_type) {
+		char buf[1024];
+		char str[UUID_STR_LEN];
+
+		ret = call_daemon(uuidd_opts.socket_path, uuidd_opts.do_type + 2, buf,
+				  sizeof(buf), &uuidd_opts.num, &err_context);
 		if (ret < 0)
 			err(EXIT_FAILURE, _("error calling uuidd daemon (%s)"),
 					err_context ? : _("unexpected error"));
 
-		if (do_type == UUIDD_OP_TIME_UUID) {
-			if (ret != sizeof(uu) + sizeof(num))
+		if (uuidd_opts.do_type == UUIDD_OP_TIME_UUID) {
+			if (ret != sizeof(uuid_t) + sizeof(uuidd_opts.num))
 				unexpected_size(ret);
 
 			uuid_unparse((unsigned char *) buf, str);
 
 			printf(P_("%s and %d subsequent UUID\n",
-				  "%s and %d subsequent UUIDs\n", num - 1),
-			       str, num - 1);
+				  "%s and %d subsequent UUIDs\n", uuidd_opts.num - 1),
+			       str, uuidd_opts.num - 1);
 		} else {
+			uint32_t i;
+
 			printf(_("List of UUIDs:\n"));
 			cp = buf + 4;
-			if (ret != (int) (sizeof(num) + num * sizeof(uu)))
+			if (ret != (int) (sizeof(uuidd_opts.num) + uuidd_opts.num * sizeof(uuid_t)))
 				unexpected_size(ret);
-			for (i = 0; i < num; i++, cp += UUID_LEN) {
+			for (i = 0; i < uuidd_opts.num; i++, cp += UUID_LEN) {
 				uuid_unparse((unsigned char *) cp, str);
 				printf("\t%s\n", str);
 			}
 		}
 		return EXIT_SUCCESS;
 	}
-	if (do_type) {
-		ret = call_daemon(socket_path, do_type, (char *) &uu,
-				  sizeof(uu), 0, &err_context);
+	if (uuidd_opts.do_type) {
+		uuid_t uu;
+		char str[UUID_STR_LEN];
+
+		ret = call_daemon(uuidd_opts.socket_path, uuidd_opts.do_type, (char *) &uu,
+				  sizeof(uuid_t), 0, &err_context);
 		if (ret < 0)
 			err(EXIT_FAILURE, _("error calling uuidd daemon (%s)"),
 					err_context ? : _("unexpected error"));
-		if (ret != sizeof(uu))
+		if (ret != sizeof(uuid_t))
```

Comment:
Again, not related to the primary patch goal (and I think sizeof(uu) is better ;-)).